src/marshmallow/fields.py

```diff
@@ -1484,49 +1491,38 @@ class TimeDelta(Field):
     def __init__(
         self,
         precision: str = SECONDS,
-        serialization_type: type[int | float] = int,
+        serialization_type: typing.Any = missing_,
         **kwargs,
-    ):
+    ) -> None:
         precision = precision.lower()
-        units = (
-            self.DAYS,
-            self.SECONDS,
-            self.MICROSECONDS,
-            self.MILLISECONDS,
-            self.MINUTES,
-            self.HOURS,
-            self.WEEKS,
-        )
 
-        if precision not in units:
-            msg = 'The precision must be {} or "{}".'.format(
-                ", ".join([f'"{each}"' for each in units[:-1]]), units[-1]
-            )
+        if precision not in self._unit_to_microseconds_mapping:
+            units = ", ".join(self._unit_to_microseconds_mapping)
+            msg = f"The precision must be one of: {units}."
             raise ValueError(msg)
 
-        if serialization_type not in (int, float):
-            raise ValueError("The serialization type must be one of int or float")
+        if serialization_type is not missing_:
```

Comment:
Since we're actually removing the serialization_type rather than deprecating, this code block becomes unnecessary

Reply:
ah right. so I'll remove? and people on v2 still passing the arg will get silently propagated into `kwargs` of super().init

Reply:
7519edc

Reply:
additional_metadata will be removed in 4.0, so it passing `serialization_type` will fail loudly (which we want)